cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -131,60 +139,69 @@ struct MurmurHash3_32 {
     return combined;
   }
 
-  result_type __device__ inline operator()(Key const& key) const { return compute(key); }
+  // TODO Do we need this operator() and/or compute? Probably not both.
```

Comment:
Also for a future PR: I would try removing the `compute` method and move its definition to `operator()`. I think we might be able to safely remove the `template <typename T>` on `compute(T)`. The `operator()` template parameter `T` has to match the class template parameter `Key`, from what I can see, and may be redundant. Any exceptions to this would probably be solved by removing the class template parameter `Key` and switching to just an `operator()` template parameter.